src/java/org/apache/cassandra/service/reads/DataResolver.java

```diff
@@ -260,7 +270,10 @@ private PartitionIterator resolveWithReplicaFilteringProtection(E replicas, Repa
                                                                  UnaryOperator.identity());
 
         PartitionIterator completedPartitions = resolveWithReadRepair(secondPhaseContext,
-                                                                      i -> rfp.queryProtectedPartitions(firstPhasePartitions, i),
+                                                                      i -> shortReadProtectedResponse(i, 
+                                                                                                      secondPhaseContext,
+                                                                                                      null,
+                                                                                                      rfp.queryProtectedPartitions(firstPhasePartitions, i)),
```

Comment:
@adelapena I'm not sure if this is the right way to add SRP to the result of post-filtering, but it does seem to have helped in my fuzz tests here...